app/Http/Controllers/Api/V1/ApiProductController.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Http\Controllers\Api\V1;
+
+use App\Http\Controllers\Controller;
+use App\Models\Product;
+use Illuminate\Http\Request;
+
+class ApiProductController extends Controller
+{
+    public function index()
+    {
+        return Product::get();
+    }
+
+    public function show(Product $product)
+    {
+        return $product;
+    }
+
+    public function store(CategoryRequest $request){
+        $product = new Product();
+        $product->name = $request->name;
+        $product->save();
+        return $product;
+    }
+
+    public function update(CategoryRequest $request, Product $product)
```

Comment:
Maybe it makes sense to create a own request for the product. (ProductRequest)
```suggestion
    public function update(ProductRequest $request, Product $product)
```